src/rqt_graph/rosgraph2_impl.py

```diff
@@ -0,0 +1,526 @@
+# Software License Agreement (BSD License)
+#
+# Copyright (c) 2008, Willow Garage, Inc.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above
+#    copyright notice, this list of conditions and the following
+#    disclaimer in the documentation and/or other materials provided
+#    with the distribution.
+#  * Neither the name of Willow Garage, Inc. nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# Brawner: Adopted from ros_comm for ROS2
+
+"""
+Data structures and library for representing ROS Computation Graph state.
+"""
+
+import sys
+import time
+import itertools
+import random
+import logging
+import traceback
+import socket
+
+from collections import defaultdict
+
+from python_qt_binding.QtCore import qDebug, qWarning
+from rclpy import logging
+logger = logging.get_logger('rqt_graph.rosgraph2_impl.graph')
+
+_ROS_NAME = '/rosviz'
+
+
+def topic_node(topic):
+    """
+    In order to prevent topic/node name aliasing, we have to remap
+    topic node names. Currently we just prepend a space, which is
+    an illegal ROS name and thus not aliased.
+    @return str: topic mapped to a graph node name.
+    """
+    return ' ' + topic
+
+
+def node_topic(node):
+    """
+    Inverse of topic_node
+    @return str: undo topic_node() operation
+    """
+    return node[1:]
+
+
+class BadNode(object):
+    """
+    Data structure for storing info about a 'bad' node
+    """
+
+    # no connectivity
+    DEAD = 0
+    # intermittent connectivity
+    WONKY = 1
+
+    def __init__(self, name, type, reason):
```

Comment:
rename `type`. 
`Flake8: A002 ""test.py:1:29: A002 "type" is used as an argument and thus shadows a python builtin, consider renaming the argument`